panic when using a database with aws_rds config

I'm getting a panic when trying to use an AWS RDS instance:

```txt
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xfea666]

goroutine 1 [running]:
github.com/styrainc/opa-control-plane/internal/config.(*Root).SetSQLitePersistentByDefault(0xc000014910, {0x2ce0000, 0x4})
        /src/internal/config/config.go:58 +0xc6
github.com/styrainc/opa-control-plane/cmd/run.init.0.func1(0xc0006e5a00?, {0x2ce00ec?, 0x4?, 0x2ce00f0?})
        /src/cmd/run/run.go:62 +0x3df
github.com/spf13/cobra.(*Command).execute(0xc000608608, {0x54b8f40, 0x0, 0x0})
        /go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1019 +0xa91
github.com/spf13/cobra.(*Command).ExecuteC(0x546efc0)
        /go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1148 +0x46f
github.com/spf13/cobra.(*Command).Execute(0xc000002380?)
        /go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1071 +0x13
main.main()
        /src/main.go:17 +0x1a
```

The config looks like this:

```yml
...
database:
  aws_rds:
    region: ${REGION}
    endpoint: ${RDS_CLUSTER_ENDPOINT}
    driver: mysql
    database_user: ${RDS_USERNAME}
    database_name: opactl
    root_certificates: /etc/ssl/certs/aws-rds-ca-bundle.pem
...
```

I suspect the error is from [this function](https://github.com/open-policy-agent/opa-control-plane/blob/1f8ce951b9b3992c6c77f373c9d94e4a04e7e1e7/internal/config/config.go#L54). Since we don't have a SQL config set, `r.Database.SQL` remains nil, so the following check against `r.Database.SQL.Driver` blows up.

--- a/internal/config/config.go
+++ b/internal/config/config.go
@@ -49,16 +49,20 @@ type Root struct {
 func (r *Root) SetSQLitePersistentByDefault(persistenceDir string) bool {
 	if r.Database == nil {
 		r.Database = &Database{}
+	} else if r.Database.AWSRDS != nil {
+		return false
 	}
 
 	if r.Database.SQL == nil && r.Database.AWSRDS == nil {
 		r.Database.SQL = &SQLDatabase{}
 	}
 
-	if (r.Database.SQL.Driver == "" ||
-		r.Database.SQL.Driver == "sqlite3" || r.Database.SQL.Driver == "sqlite") && r.Database.SQL.DSN == "" {
-		r.Database.SQL.Driver = "sqlite3"
-		r.Database.SQL.DSN = filepath.Join(persistenceDir, "sqlite.db")
+	switch r.Database.SQL.Driver {
+	case "", "sqlite3", "sqlite":
+		if r.Database.SQL.DSN == "" {
+			r.Database.SQL.Driver = "sqlite3"
+			r.Database.SQL.DSN = filepath.Join(persistenceDir, "sqlite.db")
+		}
 		return true
 	}
 	return false

--- a/internal/config/config_test.go
+++ b/internal/config/config_test.go
@@ -7,6 +7,7 @@ import (
 	"strings"
 	"testing"
 
+	"github.com/google/go-cmp/cmp"
 	"github.com/styrainc/opa-control-plane/internal/config"
 	"gopkg.in/yaml.v3"
 )
@@ -272,4 +273,68 @@ func TestTopoSortSourcesCycle(t *testing.T) {
 		t.Fatal("expected cycle error on source A but got:", err)
 	}
 
+}
+
+func TestSetSQLitePersistentByDefault(t *testing.T) {
+	for _, tc := range []struct {
+		note      string
+		input     *config.Root
+		exp       *config.Root
+		expSQLite bool
+	}{
+		{
+			note:  "no database whatsoever",
+			input: &config.Root{},
+			exp: &config.Root{Database: &config.Database{
+				SQL: &config.SQLDatabase{
+					Driver: "sqlite3",
+					DSN:    "/tmp/sqlite.db",
+				},
+			}},
+			expSQLite: true,
+		},
+		{
+			note: "existing non-sqlite SQL database",
+			input: &config.Root{Database: &config.Database{
+				SQL: &config.SQLDatabase{
+					Driver: "postgresql",
+					DSN:    "postgresql://foo:bar@localhost:5432/opactl",
+				},
+			}},
+			exp: &config.Root{Database: &config.Database{
+				SQL: &config.SQLDatabase{
+					Driver: "postgresql",
+					DSN:    "postgresql://foo:bar@localhost:5432/opactl",
+				},
+			}},
+			expSQLite: false,
+		},
+		{
+			note: "existing AWSRDS database",
+			input: &config.Root{Database: &config.Database{
+				AWSRDS: &config.AmazonRDS{
+					Driver: "mysql",
+					DSN:    "mysql://foo:bar@localhost:5432/opactl",
+				},
+			}},
+			exp: &config.Root{Database: &config.Database{
+				AWSRDS: &config.AmazonRDS{
+					Driver: "mysql",
+					DSN:    "mysql://foo:bar@localhost:5432/opactl",
+				},
+			}},
+			expSQLite: false,
+		},
+	} {
+		t.Run(tc.note, func(t *testing.T) {
+			p := "/tmp"
+			isSQLite := tc.input.SetSQLitePersistentByDefault(p)
+			if diff := cmp.Diff(tc.exp, tc.input); diff != "" {
+				t.Error("unexpected diff, (-want, +got)", diff)
+			}
+			if isSQLite != tc.expSQLite {
+				t.Errorf("unexpected return value, want %v, got %v", tc.expSQLite, isSQLite)
+			}
+		})
+	}
 }

